Library/Homebrew/readall.rb

```diff
@@ -69,10 +69,12 @@ def valid_casks?(casks, bottle_tag: nil)
       casks.each do |file|
         cask = Cask::CaskLoader.load(file)
 
-        # Fine to have missing URLs for unsupported macOS
-        macos_req = cask.depends_on.macos
-        next if macos_req&.version && Array(macos_req.version).none? do |macos_version|
-          bottle_tag.to_macos_version.public_send(macos_req.comparator, macos_version)
+        if bottle_tag
```

Comment:
What does `bottle_tag` have to do with casks?

Reply:
It seems to be a way to pass the system info, os and arch, to the `valid_casks?` method to allow us to run `brew readall` against all possible variations of a cask.

Reply:
Can we pass them individually instead? It is very confusing to simulate casks based on the bottle tag since casks don't have bottles.

Reply:
Good point, I'll see about refactoring it a bit.